From 93a1559e6041371407c0b69ffcd0fc0eb421e1d6 Mon Sep 17 00:00:00 2001
From: zhouliang2 <zhouliang2@eswincomputing.com>
Date: Fri, 27 Dec 2024 17:20:07 +0800
Subject: [PATCH 367/652] WIN2030-16898:feat:revert reboot generate noise bug

Changelogs:
1. es8328.c
2. add power control when mute to fix new borad reboot generate noise
3. but at same time, it causes no sound on repeat playbcak

Change-Id: I0fcc3bb809f0800e71ac675b45b1cd779796efd1
Signed-off-by: zhouliang2 <zhouliang2@eswincomputing.com>
---
 sound/soc/codecs/eswin/es8328.c | 8 ++++++--
 sound/soc/eswin/esw-dma.c       | 8 ++++----
 sound/soc/eswin/esw-i2s.c       | 2 +-
 3 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/sound/soc/codecs/eswin/es8328.c b/sound/soc/codecs/eswin/es8328.c
index 26aedb86a7c9..924190fcacfa 100644
--- a/sound/soc/codecs/eswin/es8328.c
+++ b/sound/soc/codecs/eswin/es8328.c
@@ -700,8 +700,12 @@ static const struct snd_soc_dapm_route es8328_dapm_routes[] = {
 
 static int es8328_mute(struct snd_soc_dai *dai, int mute, int direction)
 {
-	if ((mute == 1) && (direction == 0)) {
-		snd_soc_component_write(dai->component, ES8328_DACPOWER, 0xc0);
+	if (direction == 0) {
+		if (mute == 1) {
+			snd_soc_component_write(dai->component, ES8328_DACPOWER, 0xC0);
+		} else {
+			snd_soc_component_write(dai->component, ES8328_DACPOWER, 0x3C);
+		}
 	}
 
 	return snd_soc_component_update_bits(dai->component, ES8328_DACCONTROL3,
diff --git a/sound/soc/eswin/esw-dma.c b/sound/soc/eswin/esw-dma.c
index 387974d16270..56de39b82b76 100644
--- a/sound/soc/eswin/esw-dma.c
+++ b/sound/soc/eswin/esw-dma.c
@@ -426,10 +426,10 @@ int esw_pcm_dma_dai_register(struct i2s_dev *chip)
 	chan0 = dma_request_chan(chip->dev, channel_names0);
 	if (IS_ERR(chan0)) {
 		if (PTR_ERR(chan0) == -EPROBE_DEFER) {
-			dev_err(chip->dev, "request dma channel[%s] failed\n", channel_names0);
+			dev_warn(chip->dev, "request dma channel[%s] failed\n", channel_names0);
 			return -EPROBE_DEFER;
 		}
-		dev_err(chip->dev, "dma channel[%s] is NULL\n", channel_names0);
+		dev_warn(chip->dev, "dma channel[%s] is NULL\n", channel_names0);
 	} else {
 		chip->chan[SNDRV_PCM_STREAM_PLAYBACK] = chan0;
 		period_bytes_max = dma_get_max_seg_size(chan0->device->dev);
@@ -444,11 +444,11 @@ int esw_pcm_dma_dai_register(struct i2s_dev *chip)
 	chan1 = dma_request_chan(chip->dev, channel_names1);
 	if (IS_ERR(chan1)) {
 		if (PTR_ERR(chan1) == -EPROBE_DEFER) {
-			dev_err(chip->dev, "request dma channel[%s] failed\n", channel_names1);
+			dev_warn(chip->dev, "request dma channel[%s] failed\n", channel_names1);
 			ret = -EPROBE_DEFER;
 			goto release_buf0;
 		}
-		dev_err(chip->dev, "dma channel[%s] is NULL\n", channel_names1);
+		dev_warn(chip->dev, "dma channel[%s] is NULL\n", channel_names1);
 	} else {
 		chip->chan[SNDRV_PCM_STREAM_CAPTURE] = chan1;
 		period_bytes_max = dma_get_max_seg_size(chan1->device->dev);
diff --git a/sound/soc/eswin/esw-i2s.c b/sound/soc/eswin/esw-i2s.c
index 0d40aee31f87..eed2859387b1 100755
--- a/sound/soc/eswin/esw-i2s.c
+++ b/sound/soc/eswin/esw-i2s.c
@@ -1106,7 +1106,7 @@ static int i2s_probe(struct platform_device *pdev)
 	} else {
 		ret = esw_pcm_dma_dai_register(i2s_drvdata);
 		if (ret) {
-			dev_err(&pdev->dev, "could not register pcm: %d\n", ret);
+			dev_warn(&pdev->dev, "could not register pcm: %d\n", ret);
 			goto err_probe;
 		}
 	}
-- 
2.54.0

